src/compiler.rs

```diff
@@ -132,4 +159,45 @@ impl Compiler {
             file: out_file,
         }
     }
+
+    fn skip_compile(&self, out_file: String) -> CompileOutput {
+        CompileOutput {
+            status: ExitStatusExt::from_raw(0),
+            stdout: vec![],
+            stderr: vec![],
+            file: out_file,
+        }
+    }
+
+    fn is_source_modified(&self, test_folder: &str, in_files: &[String], out_file: &str) -> bool {
```

Comment:
Is `&self` actually necessary here? `is_source_modified` seems like a class function, like `get_paths_for`.

Reply:
You're right it's not, but to be honest calling `Compiler::function` from inside the `Compiler` impl also seems a little strange.

Reply:
`Self::function` is equivalent, if this seems better?

Reply:
It's just strange to me that you can't call it direcly from inside the impl, but I guess this is how it is with rust.

Reply:
That becomes even more confusing with `self.function` and `Self::function` :D Let's leave it as it is with `Compiler`. I pushed the change for `is_source_modified`